portal/src/graphql/adminapi/UserDetailsScreen.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useContext } from "react";
+import { useParams } from "react-router-dom";
+import {
+  Pivot,
+  PivotItem,
+  CommandBar,
+  ICommandBarItemProps,
+} from "@fluentui/react";
+import { useQuery, gql } from "@apollo/client";
+import { FormattedMessage, Context } from "@oursky/react-messageformat";
+
+import {
+  UserDetailsScreenQuery,
+  UserDetailsScreenQueryVariables,
+  UserDetailsScreenQuery_node_User,
+} from "./__generated__/UserDetailsScreenQuery";
+import NavBreadcrumb from "../../NavBreadcrumb";
+import ShowLoading from "../../ShowLoading";
+import ShowError from "../../ShowError";
+import UserDetailSummary from "./UserDetailSummary";
+import UserDetailsAccountSecurity from "./UserDetailsAccountSecurity";
+import UserDetailsConnectedIdentities from "./UserDetailsConnectedIdentities";
+import UserDetailsSession from "./UserDetailsSession";
+
+import { nonNullable } from "../../util/types";
+import { extractUserInfoFromIdentities } from "../../util/user";
+
+import styles from "./UserDetailsScreen.module.scss";
+
+interface UserDetailsProps {
+  data: UserDetailsScreenQuery_node_User | null;
+  loading: boolean;
+}
+
+function isUserDetails(value: any): value is UserDetailsScreenQuery_node_User {
```

Comment:
I don't think this function is needed? We can just check `node?.__typename === "User"` at call site?

Reply:
You are right, it successfully infer the type after this condition